src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs

```diff
@@ -1528,8 +1528,9 @@ private void Initialize(Uri connectionUri, WSManConnectionInfo connectionInfo)
             }
 
 #if UNIX
+            bool isHttps = StringComparer.OrdinalIgnoreCase.Compare("https", connectionUri.Scheme) == 0;
             // explicitly disallow Basic auth over HTTP on Unix.
-            if (connectionInfo.AuthenticationMechanism == AuthenticationMechanism.Basic && !isSSLSpecified)
+            if (connectionInfo.AuthenticationMechanism == AuthenticationMechanism.Basic && !isSSLSpecified && !isHttps)
```

Comment:
I think you can get rid of the `isHttps` variable and just use:

```c#
connectionUri.Scheme != Uri.UriSchemeHttps
```

Reply:
@SteveL-MSFT: does connectionUri guarantee that it normalizes the case of the Scheme property?

Reply:
I believe @SteveL-MSFT means:
```c#
if (connectionInfo.AuthenticationMechanism == AuthenticationMechanism.Basic && !isSSLSpecified && StringComparer.OrdinalIgnoreCase.Compare("https", connectionUri.Scheme) != 0)

Reply:
@SteveL-MSFT is correct. `Uri.Scheme` is always normalized to lowercase (see the notes on the property value here https://docs.microsoft.com/en-us/dotnet/api/system.uri.scheme?view=netcore-2.1)

Line 1531 is unneeded and on line 1533 `&& !isHttps` can be changed to `&& connectionUri.Scheme != Uri.UriSchemeHttps`

Reply:
Fixed